### DISCUSSION TITLE
Tailwind with Vue 3 purges all computed classes

### DISCUSSION THREAD

**Discussion by anthonyma94** [tailwindlabs/tailwindcss#5255] [Help] [answered]
Hi! I hope someone here can help me out with this.

I'm using Tailwind CSS with Vue 3 (using vue-cli & Composition API), and I've noticed most of my CSS is purged when building to production. The purged classes seem to originate from computed classes. For example, take the following style and component:

`app.postcss`
```css
@layer components {
  .btn {
    @apply px-5 py-3;
    &-primary {
      @apply bg-blue-500;
    }
  }
}
```

`button.vue`
```vue
<template>
  <button class="btn" :class="computedClass"></button>
</template>
<script setup>
  const props = defineProps({
    variant: {
      type: String,
      default: "primary"
    }
  });
  const computedClass = computed(() => {
    return `btn-${props.variant}`;
  });
</script>
```

The button will lose its `btn-${props.variant}` styles but retain the `btn` styles. I know that Tailwind specs say not to have string literals for classes, would that have something to do with it? I thought that only applies to classes within the template.

My purge options are the ones recommended by Tailwind (`purge: ["./src/**/*.{vue,ts}"]`)

**Comment by lukewarlow** [comment, marked as answer]
Template literals are indeed the problem you're having. Purge works by a fairly simple string search through all configured files.
Template literals will mean none of the classes are found and so they're purged.

While it's not as nice code wise, you could do something like a switch statement and return the correct full class.

Pseudo-code 
```
switch props.variant
    case "primary":
          return "btn-primary";
   case "secondary":
          return "btn-secondary";
```
This allows purge css to correctly find all of the classes.

Alternatively if the list of possible classes would make the code too messy for your liking, you could make use of class safelists. https://tailwindcss.com/docs/optimizing-for-production#safelisting-specific-classes

This would allow you to safelist all the btn variants and then they won't ever be purged.

**Reply by anthonyma94** [reply]
Thank you for your help! I looked through that along with PurgeCSS documentation and found I could use regex as a safe list pattern, it really helps to whitelist all color variants for different classes.
